### PR TITLE
Connecting the Gift Recommendations view to the Gift Detail View

### PR DESCRIPTION
**Pull request by benbiran**
- Created a gift reducer to track the currently displayed gift
- Updated the gift detail view to handle the gift details from the redux state
- Updated the Book view to conform to the data structure passed from the Thumbnail view
- Fixed the carousel slider from on the Gift Recommendation view + added the React-Slick CSS to the HTML

### CODE REVIEW

File: src/app/Actions/gift.js [hunk 1]
```diff
@@ -13,3 +13,10 @@ export function removeFriend(id) {
     id
   };
 };
+
+export function currentDisplayedGift(gift){
+  return {
```
Comment (benbiran) [inline src/app/Actions/gift.js:18]: Display current gift - action creator

File: src/app/Containers/giftDetail.jsx [hunk 2]
```diff
@@ -1,16 +1,23 @@
...
+      <Book book={this.props.gift}/>
       </div>
     );
   }
 });
 
-module.exports = GiftDetail;
+var mapStateToProps = function(state) {
+  return {
+    gift : state.gift // export the portion of the state from index.js Reducers
```
Comment (benbiran) [inline src/app/Containers/giftDetail.jsx:19]: Exposing the gift part of the state to the gift detail view

File: src/app/Containers/giftDetail.jsx [hunk 3]
```diff
@@ -1,16 +1,23 @@
...
+  }
+};
+
+export default connect(mapStateToProps)(GiftDetail);
```
Comment (benbiran) [inline src/app/Containers/giftDetail.jsx:23]: Injecting the dispatch into the Gift Detail view

File: src/app/Reducers/index.js [hunk 4]
```diff
@@ -1,10 +1,12 @@
 import { combineReducers } from 'redux';
 import friends from './userReducer'; //TODO change the path later
 import friend from './friendReducer';
+import gift from './giftReducer';
 
 const rootReducer = combineReducers({
   friends, // state for the friendsManager
-  friend
+  friend, // whose gift recommendation page we are on
+  gift
```
Comment (benbiran) [inline src/app/Reducers/index.js:9]: Updated the master reducer

File: src/app/Reducers/giftReducer.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import {FETCH_CURRENT_DISPLAYED_GIFT} from '../Constants/ActionTypes';
+
+const initialState = {};
+
+export default function userReducer(state=initialState, action) {
+  // DO NOT mutate the state, return a NEW state
+
+  switch (action.type) {
+    case FETCH_CURRENT_DISPLAYED_GIFT:
```
Comment (benbiran) [inline src/app/Reducers/giftReducer.js:9]: Defined the action for fetching the current displayed gift in the details view

File: src/index.html [hunk 6]
```diff
@@ -13,6 +13,7 @@
         <!-- font awesome from BootstrapCDN -->
         <link href="//maxcdn.bootstrapcdn.com/bootstrap/3.2.0/css/bootstrap.min.css" rel="stylesheet">
         <link href="//maxcdn.bootstrapcdn.com/font-awesome/4.1.0/css/font-awesome.min.css" rel="stylesheet">
+        <link href="https://cdnjs.cloudflare.com/ajax/libs/slick-carousel/1.3.15/slick.css" rel="stylesheet">
```
Comment (benbiran) [inline src/index.html:16]: Added CSS for the carousel view under Gift Recommendations container

Comment (benbiran) [inline src/index.html]: Added CSS for the carousel view under Gift Recommendations container

File: src/app/Components/book.jsx [hunk 7]
```diff
@@ -2,28 +2,42 @@ var React = require('react');
 
 var Book = React.createClass({
   addToList: function() {
-    console.log("add to list clicked for the item ", this.props.book[0].ItemAttributes.Title);
+    console.log("add to list clicked for the item ", this.props.book.ItemAttributes.Title);
     // send the clicked book to the server to save on the user's gift list
     // should the req object just be Book's rendered view? this.props.book[0]
   },
 
+  componenetDidMount: function() {
```
Comment (batmanimal) [inline src/app/Components/book.jsx]: @benbiran - can you commit a quick spelling fix for componentDidMount?

**Pull request merged**